Fix #2686: Aligned home Activity component Recently played story items in landscape mode.

Akshat-gour opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
This PR Fixes #2686: This includes a change in land/dimens.xml value of home_padding_start is changed, which change the configuration to landscape of Recently played story items.
## Before Screen Shot
<img src="https://user-images.githubusercontent.com/60266468/107883074-a861ac80-6f12-11eb-8b6b-169b36639dfc.png" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/108260929-ec5de700-7188-11eb-87c5-7e219126c125.jpg" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/108260951-f1bb3180-7188-11eb-8f89-6cccf02712bc.jpg" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/109397768-6e56c880-795e-11eb-93c9-6cb58189c9fa.jpg" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/109397771-7151b900-795e-11eb-96a8-e49d54659aed.jpg" width="250" />

## After Screen Shot
<img src="https://user-images.githubusercontent.com/60266468/107877834-b99bc080-6ef4-11eb-97ae-38af07ecd248.PNG" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/108261074-1ca58580-7189-11eb-9e3b-389a56d37b6d.jpg" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/108261079-1e6f4900-7189-11eb-97f9-86a00bf36064.jpg" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/109397799-947c6880-795e-11eb-864c-d37a32076125.jpg" width="250" />
<img src="https://user-images.githubusercontent.com/60266468/109397801-96dec280-795e-11eb-91a1-f61111319687.jpg" width="250" />

**Tablet**
![image](https://user-images.githubusercontent.com/60266468/109533356-2495eb80-7ae0-11eb-80ba-ec00fc367788.png)
![image](https://user-images.githubusercontent.com/60266468/109533452-42635080-7ae0-11eb-8474-f6e07298ff89.png)

<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Akshat-gour committed:
Aligned Home Activity Recently played story item

FareesHussain commented on app/src/main/res/values-land/dimens.xml:
```diff
@@ -1,7 +1,7 @@
 <resources>
   <dimen name="home_padding_end">72dp</dimen>
-  <dimen name="home_padding_start">72dp</dimen>
+  <dimen name="home_padding_start">65dp</dimen>
```
According to [Working on UI](https://github.com/oppia/oppia-android/wiki/Working-on-UI#paddingmarginview-dimensions) we recommend using dimensions which are multiples of 4

anandwana001 replied:
@Akshat-gour Conversation should be resolved only by reviewers or the comment author.

Akshat-gour replied:
Done

anandwana001 replied:
as per here as the comment author is @FareesHussain , so he can confirm the conversation and look for what changes suggested and mark the conversation resolved. 

The PR author should not mark the conversation resolved, as you @Akshat-gour are the PR author here.

Akshat-gour replied:
@FareesHussain 
please review it. I made changes as suggested

FareesHussain replied:
You can mark this resolved now.

FareesHussain reviewed (changes requested):
@Akshat-gour PTAL

Akshat-gour committed:
changed the value of home_padding_start to multiple of 4

FareesHussain reviewed (approved):
LGTM

Akshat-gour commented:
@rt4914 please review the PR

rt4914 reviewed (changes requested):
Thanks @Akshat-gour your solution seems correct but I have found this alignment difference in mocks itself so just confirming this with designer about the final UI. Till then you can work on other issues.

@mschanteltc Have a look at the below screenshot. I have following questions related to this:
1. `Recently-Played Stories` is not aligned with `All topics`.
2. `Recently-Played Stories` is not aligned with `Promoted Story Card`.
3. `All Topics` seems to be aligned with `Place Values` topic card.

Is everything in mock correct or does this needs to be changed?
<img width="300" alt="Screenshot 2021-02-15 at 5 43 32 PM" src="https://user-images.githubusercontent.com/9396084/107945596-df3cce80-6fb5-11eb-8547-5ec6ac447797.png">


Note: Open the above screenshot in separate screen for better visibility with black background.

Mock Reference: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0dbdd000-6738-4f6b-9026-0d16737b7baa/specs/

mschanteltc commented:
I changed points 1 and 2. Point 3 should be kept the same. ([mock](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0dbdd000-6738-4f6b-9026-0d16737b7baa/)) In summary, the following should apply:

1. `Recently-Played Stories` is aligned with `All topics`.
2. `Recently-Played Stories` is aligned with `Promoted Story Card`.
3. All Topics is aligned with Place Values topic card.

Akshat-gour commented:
@rt4914 PTAL

rt4914 commented:
> I changed points 1 and 2. Point 3 should be kept the same. ([mock](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0dbdd000-6738-4f6b-9026-0d16737b7baa/)) In summary, the following should apply:
> 
> 1. `Recently-Played Stories` is aligned with `All topics`.
> 2. `Recently-Played Stories` is aligned with `Promoted Story Card`.
> 3. All Topics is aligned with Place Values topic card.

Thanks @mschanteltc 

@Akshat-gour Can you please fix the overall UI so that all three cases work.

Akshat-gour commented:
@rt4914 
I guess we are talking about Home Activty in Landscape mode
I don't know about **Promoted Story Card**
On home Activity  there are only **Recently Played Story** and **All topics**

Akshat-gour commented:
@rt4914
I guess we are talking about Home Activty in Landscape mode
I don't know about Promoted Story Card
On home Activity there are only Recently Played Story and All topics

FareesHussain commented:
@Akshat-gour Promoted Story card is the same as Recently Played Story.
In this https://github.com/oppia/oppia-android/pull/2723#issuecomment-780216116 to be more specific
> 2. Recently-Played Stories is aligned with Promoted Story Card. 

Recently-Player Stories (Title) and Promoted Story Card (Items of the Horizontal Recyclerview).

Akshat-gour committed:
Merge remote-tracking branch 'upstream/develop' into aligned-recently-played-story-items

Akshat-gour committed:
aligned home Activity component All topics in land scape mode
as well as in portrait mode

Akshat-gour commented:
> > I changed points 1 and 2. Point 3 should be kept the same. ([mock](https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/0dbdd000-6738-4f6b-9026-0d16737b7baa/)) In summary, the following should apply:
> > 
> > 1. `Recently-Played Stories` is aligned with `All topics`.
> > 2. `Recently-Played Stories` is aligned with `Promoted Story Card`.
> > 3. All Topics is aligned with Place Values topic card.
> 
> Thanks @mschanteltc
> 
> @Akshat-gour Can you please fix the overall UI so that all three cases work.

Done

@rt4914 
@FareesHussain 
PTAL 
also updated the screen short

FareesHussain reviewed (approved):
LGTM

rt4914 commented:
@mschanteltc In below images what should be the alignment.
Should the last card in carousel align with the `VIEW-ALL` end. or should the left cards align with the `Stories for you / Recently Played Stories`
<img src="https://user-images.githubusercontent.com/9396084/108858770-8ce15a80-7612-11eb-90f4-cdd6367e3439.png" width="250" />
<img src="https://user-images.githubusercontent.com/9396084/108858777-8fdc4b00-7612-11eb-898b-b5833b6cd89e.png" width="400" />

mschanteltc commented:
> @mschanteltc In below images what should be the alignment.
> Should the last card in carousel align with the `VIEW-ALL` end. or should the left cards align with the `Stories for you / Recently Played Stories`
> <img alt="" width="250" src="https://user-images.githubusercontent.com/9396084/108858770-8ce15a80-7612-11eb-90f4-cdd6367e3439.png">
> <img alt="" width="400" src="https://user-images.githubusercontent.com/9396084/108858777-8fdc4b00-7612-11eb-898b-b5833b6cd89e.png">

Have them align with the first choice where "last card in carousel aligns with the `VIEW-ALL` end"

rt4914 commented:
@Akshat-gour Please fix the above suggestion too in this PR.

BenHenning committed:
Merge branch 'develop' into aligned-recently-played-story-items

Akshat-gour committed:
Merge remote-tracking branch 'upstream/develop' into aligned-recently-played-story-items

Akshat-gour committed:
Merge remote-tracking branch 'upstream/develop' into aligned-recently-played-story-items

Akshat-gour committed:
aligned last card view of recently played story with view All button

Akshat-gour committed:
Merge remote-tracking branch 'origin/aligned-recently-played-story-items' into aligned-recently-played-story-items

Akshat-gour commented:
@rt4914 
@FareesHussain 
PTAL
Also updated the screen shots

FareesHussain reviewed (approved):
LGTM, thanks.

rt4914 reviewed (changes requested):
@Akshat-gour Thanks for his but the solution works only for mobile. Please make similar changes for `Tablet` devices too.

<img src="https://user-images.githubusercontent.com/9396084/109527718-fb725c80-7ad9-11eb-8ec6-24daae7a1b48.png" width="350" />
<img src="https://user-images.githubusercontent.com/9396084/109527730-ff05e380-7ad9-11eb-885f-59357bf0c9f3.png" width="350" />

Akshat-gour committed:
Merge remote-tracking branch 'upstream/develop' into aligned-recently-played-story-items

Akshat-gour committed:
changed layout of home screen activity for tablet in portrait and land scape mode

Akshat-gour commented:
@rt4914 
@FareesHussain 
i did for tablet and updated the screen shot

Akshat-gour commented:
@rt4914
@FareesHussain
i did for tablet and updated the screen shot
PTAL

FareesHussain reviewed (approved):
LGTM, thanks

rt4914 reviewed (approved):
Nice work @Akshat-gour
